csm/cli/support_bundle/csm_bundle_generate.py

```diff
@@ -36,14 +37,23 @@ async def init(command):
         :return:
         """
         # Read Config to Fetch Log File Path
-        log_directory_path = Conf.get(const.CSM_GLOBAL_INDEX, "Log.log_path")
+        csm_log_directory_path = Conf.get(const.CSM_GLOBAL_INDEX, "Log.log_path")
+        uds_log_directory_path = Conf.get(const.CSM_GLOBAL_INDEX, "Log.uds_log_path")
+        elasticsearch_log_path = Conf.get(const.CSM_GLOBAL_INDEX, "Log.elasticsearch_log_path")
         # Creates CSM Directory
         path = command.options.get("path")
         bundle_id = command.options.get("bundle_id")
-        path = os.path.join(path, const.CSM_GLOBAL_INDEX)
-        os.makedirs(path)
+        component_name = command.options.get("component", "csm")
+        component_data = {"csm": [csm_log_directory_path],
+                          "uds": [uds_log_directory_path],
+                          "elasticsearch": [elasticsearch_log_path]}
+        temp_path = os.path.join(path, component_name)
+        os.makedirs(temp_path, exist_ok = True)
         # Generate Tar file for Logs Folder.
-        tar_file_name = os.path.join(path, f"csm_{bundle_id}.tar.gz")
-        Tar(tar_file_name).dump([log_directory_path])
-
+        tar_file_name = os.path.join(path, f"{component_name}_{bundle_id}.tar.gz")
+        if os.path.exists(component_data[component_name][0]):
```

Comment:
Need to check if all the paths in the List exists since Tar.dump can accept multiple file from different locations.
Could be written as 
`if all(map(os.path.exists, component_data[component_name])):`

Reply:
done suggested change, thanks for crisp codelet.